File: java-imperative-ms/domain/model/src/main/java/com/maocq/model/account/Account.java
package com.maocq.model.account;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.ZonedDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder(toBuilder = true)
public class Account {
    private Integer id;
    private String userId;
    private String account;
    private String name;
    private String number;
    private BigDecimal balance;
    private String currency;
    private String type;
    private String bank;
    private ZonedDateTime creationDate;
    private ZonedDateTime updateDate;
}

File: elixir-ms/lib/domain/model/account/account.ex
defmodule ElixirMs.Model.Account do

  defstruct [
    :id,
    :user_id,
    :account,
    :name,
    :number,
    :balance,
    :currency,
    :type,
    :bank,
    :creation_date,
    :update_date
  ]
end
